vllm/model_executor/layers/activation.py

```diff
@@ -127,11 +128,28 @@ def __init__(
     def forward(self, x: torch.Tensor) -> torch.Tensor:
         return self.act(x) / self.scales
 
-    def weight_loader(self, param: nn.Parameter, loaded_weight: torch.Tensor):
+    def weight_loader(self,
+                      param: nn.Parameter,
+                      loaded_weight: Optional[torch.Tensor],
+                      weight_owner: Optional[int] = None):
         param_data = param.data
         if self.input_is_parallel:
             tp_rank = get_tensor_model_parallel_rank()
             shard_size = param_data.shape[0]
+            if weight_owner is not None:
+                tp_size = get_tensor_model_parallel_world_size()
+                tp_group = get_tensor_model_parallel_group()
+                if tp_rank == torch.distributed.get_group_rank(
+                        tp_group, weight_owner):
+                    loaded_weight_list = list(
+                        torch.split(loaded_weight, shard_size, dim=0))
+                    assert len(loaded_weight_list) == tp_size
+                    assert param_data.shape == loaded_weight_list[0].shape
+                else:
+                    loaded_weight_list = None
+                torch.distributed.scatter(param_data, loaded_weight_list,
+                                          weight_owner, tp_group)
```

Comment:
It seems to me that this code pattern has been repeated a lot throughout this PR - why not making a separate helper/util function for this or am I missing something here?

Reply:
Fixed